src/mame/sinclair/specnext.cpp

```diff
@@ -0,0 +1,2757 @@
+		m_regs.nr_bb_divmmc_ep_1 = nr_wr_dat;
+		break;
+	case 0xc0:
+		m_regs.nr_c0_im2_vector = BIT(nr_wr_dat, 5, 3);
+		m_regs.nr_c0_stackless_nmi = BIT(nr_wr_dat, 3);
+		m_maincpu->nmi_stackless_w(m_regs.nr_c0_stackless_nmi);
+		m_regs.nr_c0_int_mode_pulse_0_im2_1 = BIT(nr_wr_dat, 0);
+		break;
+	case 0xc4:
+		m_regs.nr_c4_int_en_0_expbus = BIT(nr_wr_dat, 7);
+		m_regs.nr_22_line_interrupt_en = BIT(nr_wr_dat, 1);
+		break;
+	case 0xc6:
+		m_regs.nr_c6_int_en_2_654 = (BIT(nr_wr_dat, 6) << 2) | (BIT(nr_wr_dat, 5) << 1) | BIT(nr_wr_dat, 4);
+		m_regs.nr_c6_int_en_2_210 = (BIT(nr_wr_dat, 2) << 2) | (BIT(nr_wr_dat, 1) << 1) | BIT(nr_wr_dat, 0);
+		break;
+	case 0xcc:
+		m_regs.nr_cc_dma_int_en_0_7 = BIT(nr_wr_dat, 7);
+		m_regs.nr_cc_dma_int_en_0_10 = BIT(nr_wr_dat, 0, 2);
+		break;
+	case 0xcd:
+		m_regs.nr_cd_dma_int_en_1 = nr_wr_dat;
+		break;
+	case 0xce:
+		m_regs.nr_ce_dma_int_en_2_654 = BIT(nr_wr_dat, 4, 3);
+		m_regs.nr_ce_dma_int_en_2_210 = BIT(nr_wr_dat, 0, 3);
+		break;
+	case 0xd8:
+		m_regs.nr_d8_io_trap_fdc_en = BIT(nr_wr_dat, 0);
+		break;
+	default:
+		LOGWARN("wR: %X <- %x\n", nr_wr_reg, nr_wr_dat);
+		break;
+	}
+
+	m_sprites->mirror_index_w(0b111);
+	m_sprites->mirror_inc_w(0);
+}
+
+void specnext_state::nr_02_w(u8 nr_wr_dat)
+{
+	m_regs.nr_02_bus_reset = BIT(nr_wr_dat, 7);
+	m_regs.nr_02_iotrap = BIT(nr_wr_dat, 4);
+	m_regs.nr_02_generate_mf_nmi = BIT(nr_wr_dat, 3);
+	if (!m_regs.nr_02_generate_mf_nmi)
+		;
+	m_regs.nr_02_generate_divmmc_nmi = BIT(nr_wr_dat, 2);
+	if (!m_regs.nr_02_generate_divmmc_nmi)
+		;
+
+	if (BIT(nr_wr_dat, 1))
+		machine().schedule_hard_reset();
+	else if (BIT(nr_wr_dat, 0))
+	{
+		m_regs.nr_02_reset_type = (BIT(m_regs.nr_02_reset_type, 2) << 1) | BIT(m_regs.nr_02_reset_type, 1) | BIT(m_regs.nr_02_reset_type, 0);
+		machine().schedule_soft_reset();
+
+		// Load snapshot after reboot from bootrom
+		snapshot_image_device *snapshot = subdevice<snapshot_image_device>("snapshot");
+		snapshot->set_delay(attotime::zero);
+		snapshot->call_load();
+	}
+	else
+		LOGWARN("NMI? %x\n", nr_wr_dat);
+}
+
+void specnext_state::nr_07_cpu_speed_w(u8 data)
+{
+	m_regs.nr_07_cpu_speed = data;
+	m_maincpu->set_clock_scale(1 << data);
+	m_z80dma->set_clock_scale(1 << data);
+}
+
+void specnext_state::nr_14_global_transparent_rgb_w(u8 data)
+{
+	m_regs.nr_14_global_transparent_rgb = data;
+	m_global_transparent = (m_regs.nr_14_global_transparent_rgb << 1) | BIT(m_regs.nr_14_global_transparent_rgb, 1) | BIT(m_regs.nr_14_global_transparent_rgb, 0);
+	m_ula->set_global_transparent(m_global_transparent);
+	m_layer2->set_global_transparent(m_global_transparent);
+}
+
+void specnext_state::map_fetch(address_map &map)
+{
+	map(0x0000, 0xffff).lr8(NAME([this](offs_t offset)
+	{
+		if (m_divmmc_active /*&& !machine().side_effects_disabled()*/)
+		{
+			/* Happens after RW cycles. Ignoring side effects check is correct, because doesn't matter who reset this lines
+				and such approach gives better experience in debugger UI. */
```

Comment:
How is the experience “better”?  Why should having a memory window open showing the M1 cause this to happen every time the window is updated?

Reply:
Because this condition becomes false after first call.
Alternative in debugger would be sowing next command which not going to be executed next and before execution bank view is switched to some else.